src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -79,20 +66,30 @@ def load(cls):
         json_db = mgr.get_store(cls.ssodb_config_key(), None)
         if json_db is None:
             logger.debug("No DB v%s found, creating new...", cls.VERSION)
-            db = cls(cls.VERSION, '', Saml2({}))
+            db = cls(cls.VERSION, AuthType.LOCAL, Saml2({}))
             # check if we can update from a previous version database
             db.check_and_update_db()
             return db
 
-        dict_db = json.loads(json_db)  # type: dict
-        return cls(dict_db['version'], dict_db.get('protocol'),
-                   Saml2.from_dict(dict_db.get('saml2')))
+        dict_db = json.loads(json_db)  # type: Dict[str, str]
+        protocol: AuthType = AuthType(str(dict_db.get('protocol')))
+        config: SSOAuth = (Saml2.from_dict(str(dict_db.get('saml2'))) if protocol == AuthType.SAML2
+                           else OAuth2.from_dict(str(dict_db.get('oauth2'))))
```

Comment:
You could also use the `BaseAuth.from_protocol()` routing method?

Reply:
Yep, did that